presto-main/src/main/java/com/facebook/presto/server/security/LdapServerConfig.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.server.security;
+
+import io.airlift.configuration.Config;
+
+public class LdapServerConfig
+{
+    private String ldapUrl;
+    private String baseDistinguishedName;
+    private boolean authenticationEnabled;
+    private String activeDirectoryDomain;
+
+    public String getLdapUrl()
+    {
+        return ldapUrl;
+    }
+
+    public String getBaseDistinguishedName()
+    {
+        return baseDistinguishedName;
+    }
+
+    public String getActiveDirectoryDomain()
+    {
+        return activeDirectoryDomain;
+    }
+
+    public boolean getAuthenticationEnabled()
+    {
+        return authenticationEnabled;
+    }
+
+    @Config("authentication.ldap.url")
+    public LdapServerConfig setLdapUrl(String url)
+    {
+        this.ldapUrl = url;
+        return this;
+    }
+
+    @Config("authentication.ldap.base-dn")
```

Comment:
We should have separate config classes for each LDAP module. This allows proper validation. For example, with ActiveDirectory, the AD domain is required, so we can validate that using `@NotNull` in the config class. It's also not legal to use base DN with ActiveDirectory, so that config property shouldn't be available.

Take a look at `HiveConnectorFactory` for how it uses `installModuleIf` with the various security modules. We don't necessarily need full separate modules -- it's fine to declare inline as a lambda.